### PR TITLE
openthread: fixes for the CSL Receiver

### PR DESCRIPTION
**Pull request by edmont**
Update OpenThread and Zephyr with fixes for the CSL Receiver.

This fixes the issue when CSL is not working when asserts are
disabled and improves the energy efficiency by reducing the
duration of the CSL receive window from around 10 ms to less
than 1 ms.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

KRKNWK-11203

### CODE REVIEW

**Review by lmaciejonczyk — APPROVED** [review]

**Comment by greg-fer** [comment]
Changelog entry needed? If not, please remove the label.

**Review by LuDuda — APPROVED** [review]

File: west.yml [hunk 1]
```diff
@@ -134,6 +133,10 @@ manifest:
       repo-path: sdk-cjson
       path: modules/lib/cjson
       revision: 50bddddf7fb45e4ce87bc3619df06db55d181856
+    - name: openthread
+      repo-path: sdk-openthread
+      path: modules/lib/openthread
+      revision: 6e2ab99d68f5cd4c6ad5d728b7c7baadfcd22429
```
Comment (mbolivar-nordic) [inline west.yml:139]: This looks like the wrong commit. Tip is actually pointing at 3908fb878493367a23503548049d47a138193144, which is a merge commit bringing in this commit. @rlubos was the merge commit created intentionally?

Comment (mbolivar-nordic) [reply west.yml:139]: > @rlubos was the merge commit created intentionally?

I am going to assume "yes", keep this as-is, and force push tip backwards to point to this one

Comment (rlubos) [reply west.yml:139]: Actually it was, I've assumed the same process we have in `zephyrproject-rtos/openthread` for bringing upmerges (i.e. to do this with GH UI).

**Pull request merged**